src/ktopt_planning_context.cpp

```diff
@@ -137,19 +119,37 @@ void KTOptPlanningContext::solve(planning_interface::MotionPlanResponse& res)
     return;
   }
 
+  RCLCPP_INFO(getLogger(), "Setting initial guess ...");
+  // set the initial guess
+  trajopt.SetInitialGuess(trajopt.ReconstructTrajectory(result));
+
+  // add collision constraints
+  for (double s = 0.0; s <=25.0; s++)
+  {
+    trajopt.AddPathPositionConstraint(
+      std::make_shared<MinimumDistanceLowerBoundConstraint>(
+        &plant_,
+        LOWER_BOUND,
+        &plant_context_),
+      s/COLLISION_CHECK_RESOLUTION);
+  }
+  
+  // collision free trajectory
+  auto collision_free_result = Solve(prog);
```

Comment:
Just to get this right, we're running trajopt twice, first to get a first guess without collision constraints and a second time with all constraints including collision? Do you mind extending the comment describe that, if that is correct?

Reply:
Indeed.